volatility3/framework/plugins/linux/sockstat.py

```diff
@@ -0,0 +1,506 @@
+            bpfprog_aux = bpfprog.aux
+            if bpfprog_aux:
+                extended["bpf_filter_id"] = str(bpfprog_aux.id)
+                bpfprog_name = utility.array_to_string(bpfprog_aux.name)
+                if bpfprog_name:
+                    extended["bpf_filter_name"] = bpfprog_name
+
+    def _unix_sock(self, sock: objects.StructType) -> Tuple[objects.StructType, Tuple[str, str, str]]:
+        """Handles the AF_UNIX socket family
+
+        Args:
+            sock: Kernel generic `sock` object
+
+        Returns:
+            unix_sock: The kernel's `unix_sock` object
+            sock_stat: A tuple with the source, destination and state strings.
+        """
+        unix_sock = sock.cast("unix_sock")
+        state = unix_sock.get_state()
+        saddr = unix_sock.get_name()
+        sinode = unix_sock.get_inode()
+        if unix_sock.peer != 0:
+            peer = unix_sock.peer.dereference().cast("unix_sock")
+            daddr = peer.get_name()
+            dinode = peer.get_inode()
+        else:
+            daddr = dinode = ""
+
+        saddr_tag = f"{saddr} {sinode}"
+        daddr_tag = f"{daddr} {dinode}"
+        sock_stat = saddr_tag, daddr_tag, state
+        return unix_sock, sock_stat
+
+    def _inet_sock(self, sock: objects.StructType) -> Tuple[objects.StructType, Tuple[str, str, str]]:
+        """Handles the AF_INET/6 socket families
+
+        Args:
+            sock: Kernel generic `sock` object
+
+        Returns:
+            inet_sock: The kernel's `inet_sock` object
+            sock_stat: A tuple with the source, destination and state strings.
+        """
+        inet_sock = sock.cast("inet_sock")
+        saddr = inet_sock.get_src_addr()
+        sport = inet_sock.get_src_port()
+        daddr = inet_sock.get_dst_addr()
+        dport = inet_sock.get_dst_port()
+        state = inet_sock.get_state()
+
+        if inet_sock.get_family() == "AF_INET6":
+            saddr = f"[{saddr}]"
+
+        saddr_tag = f"{saddr}:{sport}"
+        daddr_tag = f"{daddr}:{dport}"
+        sock_stat = saddr_tag, daddr_tag, state
+        return inet_sock, sock_stat
+
+    def _netlink_sock(self, sock: objects.StructType) -> Tuple[objects.StructType, Tuple[str, str, str]]:
+        """Handles the AF_NETLINK socket family
+
+        Args:
+            sock: Kernel generic `sock` object
+
+        Returns:
+            netlink_sock: The kernel's `netlink_sock` object
+            sock_stat: A tuple with the source, destination and state strings.
+        """
+        netlink_sock = sock.cast("netlink_sock")
+
+        saddr_list = []
+        src_portid = f"portid:{netlink_sock.portid}"
+        saddr_list.append(src_portid)
+        if netlink_sock.groups != 0:
+            groups_bitmap = netlink_sock.groups.dereference()
+            groups_str = f"groups:0x{groups_bitmap:08x}"
+            saddr_list.append(groups_str)
+
+        daddr_list = []
+        dst_portid = f"portid:{netlink_sock.dst_portid}"
```

Comment:
Prepending `portid` or `group` suggests that this dataa isn't standalone, but will be combined into a long string later on.  Since these may be processed by other programs rather than simply output for human consumption, we should think carefully about what it's returning and how it's returning that information.